tests/conftest.py

```diff
@@ -41,6 +42,14 @@ def restore_env_variables():
     os.environ.update(env_backup)
 
 
+@pytest.fixture(scope="function", autouse=True)
+def teardown_process_group():
+    """ Ensures that the distributed process group gets closed before the next test runs. """
+    yield
+    if torch.distributed.is_available() and torch.distributed.is_initialized():
```

Comment:
I believe the `is_available` check is redundant here

Reply:
I thought so too but on older pytorch windows versions we test in our CI, we still have to check available before hand otherwise we error :)

Reply:
For example, windows pytorch < 1.8:

```
tests\accelerators\test_accelerator_connector.py:431: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pytorch_lightning\trainer\trainer.py:511: in fit
    self._run(model)
pytorch_lightning\trainer\trainer.py:819: in _run
    self._call_setup_hook(model)  # allow user to setup lightning_module in accelerator environment
pytorch_lightning\trainer\trainer.py:1088: in _call_setup_hook
    self.accelerator.barrier("pre_setup")
pytorch_lightning\accelerators\accelerator.py:414: in barrier
    self.training_type_plugin.barrier(name=name)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pytorch_lightning.plugins.training_type.ddp.DDPPlugin object at 0x000001DDFEF93EE0>
args = (), kwargs = {'name': 'pre_setup'}

    def barrier(self, *args, **kwargs) -> None:
>       if not torch_distrib.is_initialized():
E       AttributeError: module 'torch.distributed' has no attribute 'is_initialized'
```